testing/upgradetest/versions_test.go

```diff
@@ -14,6 +16,167 @@ import (
 	"github.com/elastic/elastic-agent/pkg/version"
 )
 
+func generateTestVersions(startVersion, endVersion string) ([]*version.ParsedSemVer, error) {
+	var versionStrings []string
+	start, err := version.ParseVersion(startVersion)
+	if err != nil {
+		return nil, fmt.Errorf("invalid startVersion: %w", err)
+	}
+	end, err := version.ParseVersion(endVersion)
+	if err != nil {
+		return nil, fmt.Errorf("invalid endVersion: %w", err)
+	}
+
+	if !start.Less(*end) && !start.Equal(*end) {
+		return nil, fmt.Errorf("start version %s is newer than end version %s", startVersion, endVersion)
+	}
+
+	for major := start.Major(); major <= end.Major(); major++ {
+		// Arbitrarily chosen well defined range of minor versions
+		minorStart := 0
+		minorEnd := 19
```

Comment:
nit: i'd make this configurable, for now it will probably be the same in every call